### ISSUE TITLE
double check WAM's MMS for low plasma density from recycling run compared with GloTec 

### ISSUE THREAD

**Issue by ZhuxiaoLi66** [SWPC-IPE/WAM-IPE#322]
Dom and Houjun did the comparison of TEC between Houjun's 'nowi' version recycling run and the GloTec output. The results shows that 'nowi' has consistent low TEC compared with GloTec. The works in this issue will help to figure out what is the cause for that.

**Comment by ZhuxiaoLi66** [comment]
[nwoi-glotec-12.docx](https://github.com/SWPC-IPE/WAM-IPE/files/3517081/nwoi-glotec-12.docx)
[nwoi-glotec-12.pdf](https://github.com/SWPC-IPE/WAM-IPE/files/3517090/nwoi-glotec-12.pdf)

**Comment by ZhuxiaoLi66** [comment]
Houjun compared the WAM mean mass output for his old and current runs. 

[Uploading the 'correct' image - wanghj 21/08/2019.]

![image](https://user-images.githubusercontent.com/12502567/63472783-77dda700-c430-11e9-973d-9f949288d1d5.png)

**Comment by ZhuxiaoLi66** [comment]
I did a similar WAM 'real time' run with Houjun's run, has got a different mean mass zonal mean structure as following.  
![MMS_zonal_WAM_swin_MSIS_20190515_weimer2005_JH05_5p_0UT](https://user-images.githubusercontent.com/22546571/63289090-cf9fd500-c2ad-11e9-97cd-1e42ad6bf23d.png)

The current run is with new weimer, a run with development tip version has been shot out, with check the output later. for now, we don't expect it will be very different with the new weimer run.
 
![MMS_zonal_monthly_WAM_MSIS_0516_weimer2005_1996_JH05_120_201305_5p](https://user-images.githubusercontent.com/22546571/63289302-53f25800-c2ae-11e9-886b-5e9b936b445a.png)

**Comment by akubaryk** [comment]
Going to run an identical standalone WAM on WCOSS Phase 2 to Zhuxiao's run so we can do an inter-machine comparison.

**Comment by akubaryk** [comment]
3hr restart realtime input parameter runs on Theia and WCOSS both fail shortly after fh117; a 15 day run with realtime parameters (which would be fixed after ~three days) succeeded on Theia according to @ZhuxiaoLi66.

I plotted the input parameters in 3hr cycles by minute (bad quality plot, sorry):
![image](https://user-images.githubusercontent.com/31134616/63376803-e8f75e80-c34b-11e9-8024-59564f85da55.png)
F10.7, Kp, Hemispheric Power, Bt, Solar Wind Angle, SW Velocity, Bz, SW density.

The F10.7=68 right at the end, where the failure is, caught my eye because of this bit of code in wam_f107_kp_mod.f:
```
! HW 15May2016
! In case of f107 == 0 or too small, 
! use the last-readin 'correct' value or f107_81d_avg 
      do i = 1, f107_kp_size
        if (f107_wy(i) >= 70.0) then
          f107_work = f107_wy(i)
        endif
      enddo

      if (f107_work < 70.0) f107_work = f107_81d_avg

      do i = 1, f107_kp_size
        if (f107_wy(i) < 70.0) then
          f107_wy(i) = f107_work
        endif
      enddo
```
That is sort of adequate code if f107_work is guaranteed to be an initialized value, but in the event that it's not, you wind up with f107 values that are garbage (10^34), which shatters the model.

From a test restart run for the failure point in question, I added this code to idea_phys.f:
```
if ( me == 0 ) then
  print *,'kdt',f107_curdt,f107d_curdt,kp_curdt,kpa_curdt
```
And got this output:
```
 kdt  1.991771521664903E+034   73.0156667000000        2.00000000000000
  0.743166700000000
```
F10.7 2*10^34, not good, and the model immediately crashed.

There was existing code in the realtime input parameter parsing that already made the above code block somewhat superfluous, limiting the lower bound to 68, but anything that crept in between 68 and 70 had the potential to cause a blow-up. I've removed that code block and added the F10.7 minimums to all runs, not just the realtime runs.

The fixes in `development_f107_kp_fix` result in a 3hr-repeat-restart run on Theia that passes fh117. A pull request will be opened.

**Comment by ZhuxiaoLi66** [comment]
Good catch, Adam. Going to approve your pull request. Tim, would like to get your confirm about the minimum F10.7 for all the runs, include the 'timeobs' (read all geomagnetic and solar input parameters from database files).

**Comment by wanghj2015** [comment]
Direct comparison of cycled run 'nwoi' with Zhuxiao's weimer2005 run 'zxll':='Weimer2005_SWIN'. The differences appear to be rather small, similar in magnitudes to the differnces of 'Weimer2015_SWIN - MSIS_70'.

![image](https://user-images.githubusercontent.com/12502567/63472256-c68a4180-c42e-11e9-82a9-00e0b752fd09.png)

**Comment by ZhuxiaoLi66** [comment]
@akubaryk please double check the way Houjun's recycling reading the input parameters, do you think it is very reasonable for it to repeatly reading the first 3 hrs data for all the simulation time?

**Comment by akubaryk** [comment]
I need more information; is there a path available to inspect the run scripts and see how the input parameters are being handled?

**Comment by ZhuxiaoLi66** [comment]
Maybe I got it wrong. Just wish that after we can produce the whole set of time dependent input parameters on WCOSS, Houjun's recycling system can read in the input really time 'dependently'. I mean the input can be different for each 3 hrly circle.

**Comment by akubaryk** [comment]
That is what my scripts do as well. Every three hour cycle it reads 20190515T0015Z.xml, then 20190515T0315Z.xml, etc.

**Comment by wanghj2015** [comment]
You can find the info here, for example,
/scratch4/NCEPDEV/stmp4/Houjun.Wang/prwam.nwoi/\*fcst\*dayfile, e.g., wam2019051506gdasfcst1_0.dayfile

cycle00 use:
/swpc/noscrub/prod/data_transfer_in/wam/wam_input-20190515T0015.xml
cycle06 use:
/swpc/noscrub/prod/data_transfer_in/wam/wam_input-20190515T0615.xml

But IAU starts at -3 hr for each cycle time.

On 'theia', these xml files should be in: /scratch3/NCEPDEV/swpc/data/wam

**Comment by ZhuxiaoLi66** [comment]
Thanks, Houjun. @akubaryk have you got chance to do the 11days run on WCOSS? please point out the run directory after you done. thanks.

**Comment by ZhuxiaoLi66** [comment]
Could you remind me what is the difference between these two runs?

**Comment by ZhuxiaoLi66** [comment]
Got it, thanks.

**Comment by ZhuxiaoLi66** [comment]
The following is the result of our sensitivity test on eddy mixing parameter as a backup solution to solve the lower TEC issue.
 
![MMS_kmax100_20130306-15](https://user-images.githubusercontent.com/22546571/64557883-11162400-d332-11e9-9352-a5c03915da5e.png)
![T_kmax100_201306-15](https://user-images.githubusercontent.com/22546571/64557896-170c0500-d332-11e9-899f-6a04f48870f3.png)

**Comment by ZhuxiaoLi66** [comment]
The global number density by the current merge version and kmax=100 case during quite time has been compared with corresponding MSIS output. 

![global_number_density_20130306-15_merge_tip](https://user-images.githubusercontent.com/22546571/65446042-0922ad80-de23-11e9-9724-a7ae303c4046.png)

![global_number_density_20130306-15_kmax100_0101](https://user-images.githubusercontent.com/22546571/65446053-0e7ff800-de23-11e9-895f-2cae07dfcd54.png)

**Comment by ZhuxiaoLi66** [comment]
![MMS_kmax180_201303](https://user-images.githubusercontent.com/22546571/68963074-fcd92200-07cd-11ea-95c9-f898e453e916.png)

**Comment by timfullerrowell** [comment]
Thanks Zhuxiao,

A nice clear illustration of the effect and the latitude dependence in the
response, which is interesting.

I was expecting the low latitude to respond more, but I think what's
happening is that high latitude heating just becomes more significant when
the background temperature are reduced by eddy mixing.

Would you be able to do the equivalent plots for temperature.

Thanks,
Tim

**Comment by ZhuxiaoLi66** [comment]
Thanks, Tim. Yes, it is different from the kmax=100 case (shown in plots in former panels). 
The the following is the corresponding zonal temperature plot. 
![Temp_zonal_kmax_20130306-0315_](https://user-images.githubusercontent.com/22546571/69077037-a3baf980-0a2c-11ea-9a38-a8c2d242b8f7.png)

**Comment by timfullerrowell** [comment]
Umm..
Thanks Zhuxiao

It looks like it gets warmer with increased eddy diffusion.
Opposite of what we were expecting.
Could you double check the labels.
Thanks,
Tim

**Comment by ZhuxiaoLi66** [comment]
Tim, It seems the labels are right... the temperature for the kmax=100 case is also kind of cooler then kamx=140 (see plot above)... don't know why.

**Comment by timfullerrowell** [comment]
Thanks Zhuxiao,
Pity.
Everything else was identical for the two runs, F10 etc?
Tim

**Comment by ZhuxiaoLi66** [comment]
should be the same, will double check.

**Comment by ZhuxiaoLi66** [comment]
The inconsistence of the change tendency in WAM between the temperature and the kmax value in idea-tracer.f has been figured out by Tim and me. It is due to the difference in Keddy related formular in idea_tracer.f (composition related) and idea_dissipation.f (temperature related). already put it on our future task list, for now, it is not influence the WAM-IPE TEC tuning result much.
